src/parsemultipart.jl

```diff
@@ -0,0 +1,52 @@
+function parse_multipart_chunk!(chunk, d)
+    re = r"(\n){2,}|(\r){2,}|[\r\n]{4,}"
+    Parsers.exec(re, chunk) || return nothing
+    i, j = re.ovec
+    description = String(chunk[1:i])
+    content = Parsers.nextbytes(re, chunk)
+
+    v = match(r"Content-Disposition: form-data; name=\"(.*)\"; filename=\"(.*)\"[\r\n]+Content-Type: (\S*)", description)
+    if v !== nothing
+        push!(d["name"], v[1])
+        push!(d["filename"], v[2])
+        push!(d["contenttype"], v[3])
+    else
+        v = match(r"Content-Disposition: form-data; name=\"(.*)\"", description)
+        v == nothing && return
+        push!(d["name"], v[1])
+        push!(d["filename"], nothing)
+        push!(d["contenttype"], nothing)
+    end
+
+    re = r"(?:[\r\n]*)(?:-*)$"
+    Parsers.exec(re, content) && (content = content[1:re.ovec[1]])
+    push!(d["content"], content)
+end
+
+parse_multipart_body(body::AbstractArray{UInt8}, boundary) =
+    parse_multipart_body(String(body), boundary)
+
+function parse_multipart_body(body, boundary)
+    body = String(body)
+    dict = Dict(
+        "name" => String[],
+        "filename" => Union{String, Void}[],
```

Comment:
Use `Nothing` instead of `Void`

Reply:
I see, so I should be coding with just Julia 0.7 in mind?

Reply:
Definitely. HTTP.jl has stayed master-compatible since 0.6 branched.

Reply:
I would say that at this point, with 0.7 beta out, feature addition PRs for HTTP.jl could(should?) be 0.7-only. i.e. we should put `julia 0.7` in `REQUIRE` soon anyway.